addons/l10n_in/models/account_invoice.py

```diff
@@ -97,11 +97,16 @@ def _post(self, soft=True):
             """Check state is set in company/sub-unit"""
             company_unit_partner = move.journal_id.l10n_in_gstin_partner_id or move.journal_id.company_id
             if not company_unit_partner.state_id:
-                raise ValidationError(_(
-                    "State is missing from your company/unit %(company_name)s (%(company_id)s).\nFirst set state in your company/unit.",
-                    company_name=company_unit_partner.name,
-                    company_id=company_unit_partner.id
-                ))
+                msg = _("Your company %s need to have a correct address in order to validate this invoice.\n"
+                "Set the address of your company (Don't forget the State field)") % (company_unit_partner.name)
+                action = {
+                    'view_mode': 'form',
+                    'res_model': 'res.company',
+                    'type': 'ir.actions.act_window',
+                    'res_id' : move.journal_id.company_id.id,
```

Comment:
```suggestion
                    'res_id' : move.company_id.id,
```

Reply:
@jco-odoo  I have applied the changes suggested by you.

Reply:
@jco-odoo , any update on this??